core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -274,25 +277,28 @@ private[spark] object JettyUtils extends Logging {
       conf: SparkConf,
       serverName: String = ""): ServerInfo = {
 
-    val collection = new ContextHandlerCollection
     addFilters(handlers, conf)
 
     val gzipHandlers = handlers.map { h =>
+      h.setVirtualHosts(Array("@" + SPARK_CONNECTOR_NAME))
```

Comment:
`addHandler` here means `ServerInfo#addHandler`, sorry for confusing.
And I noticed `serverInfo` in `WebUI` can be `None` and `serverInfo.foreach(_.addHandler(handler))` is not called within `WebUI#attachHandler` in that case. So I understand it's reasonable change.
Can you add a test case for justification of this change?